lib/cn/kvset.c

```diff
@@ -2350,32 +2350,32 @@ static void
 kblk_start_read(struct kvset_iterator *iter, struct kblk_reader *kr, enum read_type read_type)
 {
     bool success       HSE_MAYBE_UNUSED;
-    struct kvset_kblk *kblk;
 
     assert(!kr->mbio.pending);
     assert(iter->workq);
 
     if (kr->kr_nodex == kr->kr_nodec) {
         struct wbt_desc *wbt;
 
-        if (kr->kr_next_kblk_idx == kr->kr_kblk_cnt) {
+        if (kr->kr_next_blk_idx == kr->kr_blk_cnt) {
```

Comment:
I like to use >= in cases like this to be somewhat more defensive, then you could drop an assert(blk_idx == blk_cnt) inside the if block to be extra cautious...